src/host/linalg.jl

```diff
@@ -204,13 +204,14 @@ LinearAlgebra.permutedims!(dest::AbstractGPUArray, src::AbstractGPUArray, perm)
 ## norm
 
 function LinearAlgebra.norm(v::AbstractGPUArray{T}, p::Real=2) where {T}
-    if p == Inf
+    norm_x = if p == Inf
         maximum(abs.(v))
     elseif p == -Inf
         minimum(abs.(v))
     else
-        mapreduce(x->abs(x)^p, +, v; init=zero(T))^(1/p)
+        mapreduce(x->abs(x)^p, +, v; init=zero(T)/2)^(1/p)
```

Comment:
Note: This "/2" just enforce the init to be of expected result type to avoid odd corner cases (such as norm of Vector{Int} that is Float not Int).

Reply:
That seems like a weird way to do that. How does Base's implementation get to that type? I see a `promote_type(Float64, T)` in `generic_norm2`.

Reply:
Base seems to always compute the norm to FLoat64 (therefore the `promote_type`) then reconvert it to input type if needed. I can look into it but would be a bit more intricated. This was the "simplest" way I could think of but can work on cleaner way.

Reply:
Switched to `float(zero(T))` to match `Base`. Seems to produce to correct expected type.